src/pci_lmt/collector.py

```diff
@@ -122,28 +139,14 @@ def collect_lane_margin_on_device_list(
                     device_info=dev.device_info,
                     lane=lane,
                     receiver_number=ty.cast(int, self.receiver_number),
-                    step=steps,
+                    step=self.step,
                 )
 
-                if voltage_or_timing == "TIMING":
-                    # FIXME: move this string construction in a separate method; or better make it
-                    # into a typed enum
-                    lane_result.margin_type = "timing_"
-                    if left_right_none == 0:
-                        lane_result.margin_type += "right"
-                    elif left_right_none == 1:
-                        lane_result.margin_type += "left"
-                    else:
-                        lane_result.margin_type += "none"
+                if self.margin_type == "TIMING":
+                    lane_result.margin_type = get_timing_margin_type_as_str()
```

Comment:
i suggest using proper types and the type checker instead of labeling methods with types (here `x_as_str()`)

Reply:
Thanks for the approval. Will address in the next PR (need some time to learn more about the enum types and how it can be used here)

Reply:
that too, yes; the string `"TIMING"` should be an enum type